src/hotspot/share/gc/g1/g1Policy.cpp

```diff
@@ -1380,7 +1380,7 @@ size_t G1Policy::allowed_waste_in_collection_set() const {
   return G1HeapWastePercent * _g1h->capacity() / 100;
 }
 
-uint G1Policy::calc_min_old_cset_length(G1CollectionSetCandidates* candidates) const {
+uint G1Policy::calc_min_old_cset_length(uint num_marked_regions) const {
```

Comment:
I actually think the previous signature is easier to read; The name, `num_marked_regions`, doesn't tell me it's in fact the number of candidates.

Reply:
Later there will be a distinction within the candidates between regions added by marking, and regions added by evacuation failure/pinning. This is a very important distinction for this method to keep mixed gcs working as they are. The comment explains it. I would prefer to not change the signature again a few PRs later.

As for the previous signature being easier to read, I think keeping the information passed to method as succinct as possible makes it easier to reason about what the method actually does (and needs), so I would like to keep it as is. We can certainly argue about the actual naming (`num_marked_regions` needs a bit of an explanation as done in the comment), but I could not find something better.
I.e. `f(int) -> int` is imo easier than `f(class with lots of stuff inside) -> int`.

Reply:
> Later there will be a distinction

I think one can cross that bridge when we get there. Given limited info in this PR, it's confusing to highlight the diff prematurely, IMO.